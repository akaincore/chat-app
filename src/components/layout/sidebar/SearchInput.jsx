import React from 'react';
import { withStyles } from '@material-ui/core/styles';
import Input from '@material-ui/core/Input';

const styles = theme => ({
  container: {
    display: 'flex',
    justifyContent: 'center',
    flexWrap: 'wrap',
    padding: '10px',
  },
  input: {
    margin: theme.spacing.unit,
  },
});

const SearchInput = ({ classes, value, onSearch }) => (
  <div className={classes.container}>
    <Input
      fullWidth
      placeholder="Search chats..."
      className={classes.input}
      value={value}
      onChange={onSearch}
      inputProps={{
        'aria-label': 'Description',
      }}
    />
  </div>
);

export default withStyles(styles)(SearchInput);
