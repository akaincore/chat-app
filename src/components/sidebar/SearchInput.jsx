import React from 'react';
import {withStyles} from '@material-ui/core/styles';
import Input from '@material-ui/core/Input';

const styles = theme => ({
  container: {
    display: 'flex',
    justifyContent: 'center',
    flexWrap: 'wrap',
    padding: '10px',
  },
  input: {
    margin: theme.spacing.unit,
  },
});

const SearchInput = props => {
  const {classes} = props;
  return (
    <div className={classes.container}>
      <Input
        fullWidth={true}
        placeholder="Search chats..."
        className={classes.input}
        inputProps={{
          'aria-label': 'Description',
        }}
      />
    </div>
  );
};

export default withStyles(styles)(SearchInput);
