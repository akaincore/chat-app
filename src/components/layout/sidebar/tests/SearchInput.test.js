/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import SearchInput from '../SearchInput';

const mockProps = {
  onSearch: jest.fn(),
  value: 'foo',
};

describe('<SearchInput />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<SearchInput {...mockProps} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders correctly', () => {
    const tree = renderer.create(<SearchInput {...mockProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
