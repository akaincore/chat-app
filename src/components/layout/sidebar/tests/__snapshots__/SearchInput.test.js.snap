// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchInput /> renders correctly 1`] = `
<div
  className="SearchInput-container-33"
>
  <div
    className="MuiInputBase-root-48 MuiInput-root-35 MuiInput-underline-39 MuiInputBase-fullWidth-57 MuiInput-fullWidth-42 SearchInput-input-34"
    onClick={[Function]}
  >
    <input
      aria-label="Description"
      className="MuiInputBase-input-58 MuiInput-input-43"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Search chats..."
      type="text"
      value="foo"
    />
  </div>
</div>
`;
